upgrade/pack_builder_instantiations/UpgradePackBuilderFacade.cpp

```diff
@@ -51,12 +59,42 @@ namespace main_
         builder.WriteUpgradePack(outputFilename, fileSystem);
     }
 
+    infra::Optional<uint8_t> UpgradePackBuilderFacade::GetOrder(const std::string& targetName, const std::map<uint8_t, std::vector<std::string>>& orderedTargets) const
+    {
+        for (const auto& targets : orderedTargets)
+        {
+            const auto targetPos = std::find(targets.second.begin(), targets.second.end(), targetName);
+            if (targetPos != targets.second.end())
+                return infra::MakeOptional(targets.first);
+        }
+        return infra::none;
+    }
+
+    bool UpgradePackBuilderFacade::CheckIfTargetIsInOrder(const std::string& target, const application::SupportedTargets& supportedTargets)
```

Comment:
This function does two things: First, it inspects the order and finds out if a target obeys the rules; second, it updates the current order. Please separate those two things. If you separate them, then you can make the function that returns a bool a const function

Reply:
Yes, Changed.